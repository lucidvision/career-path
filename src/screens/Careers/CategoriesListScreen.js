import React, { Component } from 'react'
import { FlatList, StyleSheet, View } from 'react-native'
import { ListItem } from 'components'
import jobs from 'data/jobs'

class CategoriesListScreen extends Component {
  handleCategoryPressed = (jobs) => {
    this.props.navigation.navigate('CareersList', jobs)
  }
  renderItem = ({item}) => {
    return (
      <ListItem item={item} onItemPressed={this.handleCategoryPressed} />
    )
  }
  render () {
    return (
      <View style={styles.container}>
        <FlatList
          data={jobs}
          renderItem={this.renderItem}
          keyExtractor={(item) => item.name} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})

export default CategoriesListScreen
