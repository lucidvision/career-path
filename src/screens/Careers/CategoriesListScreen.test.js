import React from 'react'
import { shallow } from 'enzyme'
import CategoriesListScreen from './CategoriesListScreen'

const createCategoriesListScreen = (override) => {
  const props = {...override}

  return shallow(<CategoriesListScreen {...props} />)
}

describe('CategoriesListScreen', () => {
  const component = createCategoriesListScreen()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('renderItem', () => {
    const renderItem = shallow(component.instance().renderItem({item: {name: 'Doctor'}}))

    it('returns the search result item', () => {
      expect(renderItem.find('Text').props().children).toBe('Doctor')
    })
  })

  describe('handleCategoryPressed', () => {
    it('navigates to job screen', () => {
      const navMock = jest.fn()
      const component = createCategoriesListScreen({
        navigation: {
          navigate: navMock
        }
      })
      component.instance().handleCategoryPressed([{name: 'Doctor'}])
      expect(navMock).toHaveBeenCalledWith('CareersList', [{name: 'Doctor'}])
    })
  })
})
