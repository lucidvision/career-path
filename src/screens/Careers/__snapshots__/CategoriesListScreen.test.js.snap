// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesListScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "jobs": Array [
            Object {
              "educationRequired": "MD",
              "name": "Doctor",
              "salary": 100000,
              "score": 2,
            },
            Object {
              "educationRequired": "Bachelors",
              "name": "Nurse",
              "salary": 50000,
              "score": 1,
            },
            Object {
              "educationRequired": "Bachelors",
              "name": "Therapist",
              "salary": 50000,
              "score": 0,
            },
          ],
          "name": "Medical",
        },
        Object {
          "jobs": Array [
            Object {
              "educationRequired": "High School",
              "name": "Tour Guide",
              "salary": 30000,
              "score": 0,
            },
            Object {
              "educationRequired": "Bachelors",
              "name": "Agent",
              "salary": 50000,
              "score": 0,
            },
            Object {
              "educationRequired": "Bachelors",
              "name": "Hotel Manager",
              "salary": 80000,
              "score": 1,
            },
          ],
          "name": "Tourism",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
